features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreViewModel.kt

```diff
@@ -653,4 +653,12 @@ class ExploreViewModel @Inject constructor(
     var syncStatus: LiveData<Resource<Double>> = MediatorLiveData<Resource<Double>>().apply {
         addSource(syncStatusService.getSyncProgressFlow().asLiveData(), this::setValue)
     }
+
+    // this is a hack. it was added to clear the exception so the SearchFragment
+    // would stop displaying a Red Error
+    fun clearSyncStatus() {
+        viewModelScope.launch {
+            syncStatusService.setSyncProgress(100.0)
+        }
+    }
```

Comment:
@Syn-McJ I created this function to clear exceptions from the status flow, so that UI would not continue to display an error.

This is not the correct way to do it -- the data shouldn't be changed, rather the UI should instead ignore the error.

Why even do this?

The sync operation happens when the app starts and if the network is offline, the sync operation will fail and the exception goes into the status flow.

When the user goes to Search in Explore Dash, the ViewModel will determine the status and find the exception.  There is a timer that tells the Red error to disappear in 15 seconds.  However, if the user leaves and returns to the Search, the same red error appears about a network error.

During this time the network may again be online, but the app does not attempt to resume sync -- so the error state persists until the app is restarted.

Perhaps a red error message for network offline should be put in a different story and PR that handles network interruptions.

Reply:
Agree about handling network interruptions. Here, we can perhaps have a `syncStatusService.clear()` method or something similar to make it less hacky.